cdap-ui/app/cdap/components/Card/index.js

```diff
@@ -13,52 +13,137 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
+
+/*
+  Usage:
+    <Card
+      header={<div>Header</div>}
+      title='some title'
+      body={<div>Body</div>} | 'some string'
+      footer={<div>Footer</div>}
+      cardClass='custom-class-name'
+      size='SM | MD | LG'
+      cardStyle={{width: '50%';}}
+      closeable
+      onClose={this.onCloseFunction.bind(this)}
+    >
+      <div>
+        the content in here will overwrite the body prop
+      </div>
+    </Card>
+*/
+
 import React, {Component, PropTypes} from 'react';
-var classNames = require('classnames');
 require('./Card.less');
+
+var classNames = require('classnames');
+
 export default class Card extends Component {
-  constructor(props) {
-    super(props);
-    this.props = props;
+  getTitle () {
+    let closeButton;
+    if (this.props.closeable) {
+      closeButton = (
+        <span
+          className='fa fa-times'
+          onClick={this.props.onClose}
+        >
+        </span>
+      );
+    }
+
+    const titleElem = (
```

Comment:
I am a little confused here. We have the function as getTitle but we are constructing the header here and if there is title we are returning the header. And I guess we are doing a similar thing in getHeader function too.  May be getHeader and getTitle could be merged into one where if we have a title then use one template otherwise use another template for the header. 

Thoughts?